Lib/http/server.py

```diff
@@ -686,10 +687,30 @@ def send_head(self):
         except OSError:
             self.send_error(HTTPStatus.NOT_FOUND, "File not found")
             return None
+
+        fs = os.fstat(f.fileno())
+            
         try:
+            # Use browser cache if possible
+            if ("If-Modified-Since" in self.headers
+                    and "If-None-Match" not in self.headers):
+                # compare If-Modified-Since and time of last file modification
+                ims = email.utils.parsedate_to_datetime(
+                    self.headers["If-Modified-Since"])
+                if (ims is not None and
+                        ims.tzinfo is datetime.timezone.utc):
```

Comment:
Maybe you should also allow tzinfo to be None, to support the obsolete “asctime” format from <https://tools.ietf.org/html/rfc7231#section-7.1.1.1> without a timezone (e.g. Sun Nov  6 08:49:37 1994)